client/src/components/sort-button.tsx

```diff
@@ -0,0 +1,109 @@
+import { useState } from "react";
+import { Button } from "./ui/button";
+import { Popover, PopoverContent, PopoverTrigger } from "./ui/popover";
+import { cn } from "~/lib/utils";
+import { Typography } from "./typography";
+import SortIcon from "~/icons/sort.icon";
+import XIcon from "~/icons/x.icon";
+import CheckIcon from "~/icons/check.icon";
+import { useRouter, useSearchParams } from "next/navigation";
+import { getSortBy } from "~/lib/get-sort-by-from-param";
+import type { SortOptionKey } from "~/lib/types";
+
+export const SortOptions: Record<SortOptionKey, string> = {
+  a_z: "Name: A to Z",
+  z_a: "Name: Z to A",
+  errors: "Errors to Valid",
+  success: "Valid to Errors",
+};
+
+const SortItem = ({
+  children,
+  active,
+  onClick,
+}: {
+  children: string;
+  active: boolean;
+  onClick: () => void;
+}) => {
+  return (
+    <Button
+      variant={"ghost"}
+      size={"sm"}
+      className={cn(
+        "flex w-full flex-row items-center justify-between text-left",
+        active && "font-bold",
+      )}
+      onClick={onClick}
+    >
+      <Typography className="text-left">{children}</Typography>
+      {active ? <CheckIcon className="mx-[3px]" /> : <div className="w-10" />}
+    </Button>
+  );
+};
+
+export default function SortButton() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const sortBy = searchParams.get("sort");
+  const [open, setOpen] = useState<boolean>(false);
+  const [active, setActive] = useState<SortOptionKey | null>(getSortBy(sortBy));
```

Comment:
Your sort state is already managed by the searchparam so you don't need another useState to set the active sort option key. updating the url searchparam and pushing the new url does that already
So you can simply store the active constant like this
```suggestion
  const active = getSortBy(sortBy);
```
Dont forget to delete the setActive(*) calls as well